src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,30 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = ();
```

Comment:
Ah, I see-- yeah, you could just backwards-compatibly change your custom type into a type alias to `!`. The reason I prefer `!` is that it has coercions that custom enums don't have, although I suppose those might be added someday.